### PR TITLE
Fix torch examples and tests that are broken by incompatible torchmetrics accuracy API change

### PR DESCRIPTION
**Pull request by WeichenXu123**
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Fix torch examples / tests that are broken by imcompatible torchmetrics accuracy API change
See https://www.google.com/url?q=https://github.com/Lightning-AI/metrics/pull/1252&sa=D&source=docs&ust=1669951711678224&usg=AOvVaw04EDZusrH6-foBNTEa426S

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [x] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
Can we also fix cross version tests?

https://github.com/mlflow/mlflow/actions/runs/3592928194/jobs/6049263830#step:13:681

File: tests/pytorch/iris.py [hunk 1]
```diff
@@ -6,23 +6,32 @@
...
 
-# NB: Older versions of PyTorch Lightning define native APIs for metric computation,
-# (e.g., pytorch_lightning.metrics.Accuracy), while newer versions rely on the `torchmetrics`
-# package (e.g. `torchmetrics.Accuracy)
-try:
-    from torchmetrics import Accuracy
-except ImportError:
-    from pytorch_lightning.metrics import Accuracy
+
+def create_multitask_accuracy():
```
Comment (harupy) [inline tests/pytorch/iris.py:12]:
```suggestion
def create_multiclass_accuracy():
```

File: tests/pytorch/iris.py [hunk 2]
```diff
@@ -6,23 +6,32 @@
...
 import torch.nn.functional as F
+from packaging.version import Version
 
-# NB: Older versions of PyTorch Lightning define native APIs for metric computation,
-# (e.g., pytorch_lightning.metrics.Accuracy), while newer versions rely on the `torchmetrics`
-# package (e.g. `torchmetrics.Accuracy)
-try:
-    from torchmetrics import Accuracy
-except ImportError:
-    from pytorch_lightning.metrics import Accuracy
```
Comment (harupy) [inline tests/pytorch/iris.py:16]: Do we need to fix this block?

Comment (harupy) [reply tests/pytorch/iris.py:16]: Can we just define 

```python
def create_multiclass_accuracy()
    if Version(torchmetrics.__version__) >= Version("0.11"):
        return Accuracy(task="multiclass")
    else:
        return Accuracy()
```

File: tests/pytorch/iris.py [hunk 3]
```diff
@@ -6,23 +6,32 @@
...
+    # package (e.g. `torchmetrics.Accuracy)
+    try:
+        from torchmetrics import Accuracy
+        import torchmetrics
+        if Version(torchmetrics.__version__) >= Version("0.11"):
+            return Accuracy(task="multiclass")
+        else:
+            return Accuracy()
+    except ImportError:
+        from pytorch_lightning.metrics import Accuracy
```
Comment (harupy) [inline tests/pytorch/iris.py:24]:
```suggestion
        from pytorch_lightning.metrics import Accuracy

```

lint error

File: tests/pytorch/iris.py (hunk 2 above)
Comment (WeichenXu123) [reply tests/pytorch/iris.py:16]: I moved it into `create_multiclass_accuracy` , shall it be put in top level ?

Comment (harupy) [reply tests/pytorch/iris.py:16]:
```python
def create_multiclass_accuracy():
    # NB: Older versions of PyTorch Lightning define native APIs for metric computation,
    # (e.g., pytorch_lightning.metrics.Accuracy), while newer versions rely on the `torchmetrics`
    # package (e.g. `torchmetrics.Accuracy)
    try:
        from torchmetrics import Accuracy
        import torchmetrics
        if Version(torchmetrics.__version__) >= Version("0.11"):
            return Accuracy(task="multiclass")
        else:
            return Accuracy()
    except ImportError:
        from pytorch_lightning.metrics import Accuracy
        return Accuracy()
```

vs.

```python
try:
    from torchmetrics import Accuracy
except ImportError:
    from pytorch_lightning.metrics import Accuracy

def create_multiclass_accuracy()
    if Version(torchmetrics.__version__) >= Version("0.11"):
        return Accuracy(task="multiclass")
    else:
        return Accuracy()
```

Comment (harupy) [reply tests/pytorch/iris.py:16]: To me, the second one looks cleaner.

**Comment by WeichenXu123** [comment]
@harupy 

Can we add a label "disableCI" to temporary disable CI when pushing updates ? Some tiny updates we don't need rerun CI otherwise it generate too many action backlog tasks and make other PR's CI waiting too long to run.

File: tests/pytorch/iris.py (hunk 2 above)
Comment (harupy) [reply tests/pytorch/iris.py:16]: Does the second one work?

**Comment by harupy** [comment]
When you push a new commit in the RP, the existing CI jobs will be canceled.

**Comment by harupy** [comment]
> when pushing updates

Pushing update where? master? PR?

File: tests/pytorch/iris.py (hunk 2 above)
Comment (harupy) [reply tests/pytorch/iris.py:16]: Perhaps the first one might be less confusing.

**Review by harupy — APPROVED** [review]
LGTM!

File: tests/pytorch/iris.py [hunk 4]
```diff
@@ -6,23 +6,34 @@
...
+    # (e.g., pytorch_lightning.metrics.Accuracy), while newer versions rely on the `torchmetrics`
+    # package (e.g. `torchmetrics.Accuracy)
+    try:
+        from torchmetrics import Accuracy
+        import torchmetrics
+
+        if Version(torchmetrics.__version__) >= Version("0.11"):
+            return Accuracy(task="multiclass")
+        else:
+            return Accuracy()
```
Comment (harupy) [inline tests/pytorch/iris.py:23]:
```suggestion
            return Accuracy()  # pylint: disable=no-value-for-parameter
```

File: examples/pytorch/AxHyperOptimizationPTL/iris.py [hunk 5]
```diff
@@ -17,9 +17,9 @@ class IrisClassification(pl.LightningModule):
     def __init__(self, **kwargs):
         super().__init__()
 
-        self.train_acc = Accuracy()
-        self.val_acc = Accuracy()
-        self.test_acc = Accuracy()
+        self.train_acc = Accuracy(task="multiclass", num_classes=10)
```
Comment (harupy) [inline examples/pytorch/AxHyperOptimizationPTL/iris.py:20]: Why 10?
